examples/scenes/src/cosmic_text_scene.rs

```diff
@@ -0,0 +1,452 @@
+
+        // Draw the Glyphs
+        for buffer_line in &buffer_glyphs.buffer_lines {
+            for glyph_highlight in &buffer_line.glyph_highlights {
+                scene.fill(
+                    Fill::NonZero,
+                    text_transform,
+                    buffer_glyphs.glyph_highlight_color,
+                    None,
+                    glyph_highlight,
+                );
+            }
+
+            if let Some(cursor) = &buffer_line.cursor {
+                scene.fill(
+                    Fill::NonZero,
+                    text_transform,
+                    buffer_glyphs.cursor_color,
+                    None,
+                    cursor,
+                );
+            }
+
+            for glyph_run in &buffer_line.glyph_runs {
+                let font = vello_fonts.get(&glyph_run.font).unwrap();
+                let glyph_color = glyph_run.glyph_color;
+                let glyphs = glyph_run.glyphs.clone();
+                scene
+                    .draw_glyphs(font)
+                    .font_size(buffer_glyphs.font_size)
+                    .brush(glyph_color)
+                    .transform(text_transform)
+                    .draw(Fill::NonZero, glyphs.into_iter());
+            }
+        }
+    }
+}
+
+pub struct CosmicTextSceneState {
+    pub font_system: FontSystem,
+    pub vello_fonts: HashMap<ID, Font>,
+    pub buffer_glyphs: BufferGlyphs,
+}
+
+impl Default for CosmicTextSceneState {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+struct CosmicFontBlobAdapter {
+    font: Arc<cosmic_text::Font>,
+}
+
+/// Adapter to allow `cosmic_text::Font` to be used as a Blob.
+impl CosmicFontBlobAdapter {
+    fn new(font: Arc<cosmic_text::Font>) -> Self {
+        Self { font }
+    }
+}
+
+impl AsRef<[u8]> for CosmicFontBlobAdapter {
+    fn as_ref(&self) -> &[u8] {
+        self.font.data()
+    }
+}
+
+impl CosmicTextSceneState {
+    pub fn new() -> Self {
+        let mut font_system = FontSystem::new();
+        let mut vello_fonts = HashMap::new();
+
+        // Copy fonts from cosmic_text, so vello can use them
+        let font_faces: Vec<(ID, u32)> = font_system
+            .db()
+            .faces()
+            .map(|face| (face.id, face.index))
+            .collect();
+
+        for (font_id, index) in font_faces {
+            if let Some(font) = font_system.get_font(font_id) {
+                // For now use an adapter, to avoid cloning the entire font data.
+                // For alternatives, see https://github.com/linebender/vello/pull/739#discussion_r1858293718
+                let font_blob = Blob::new(Arc::new(CosmicFontBlobAdapter::new(font)));
+                let vello_font = Font::new(font_blob, index);
+                vello_fonts.insert(font_id, vello_font);
+            }
+        }
+
+        let text = "おはよう (ja) (ohayō) 🌅✨ (morning), こんにちは (ja) (konnichi wa) ☀️😊 (daytime), こんばんは (ja) (konban wa) 🌙🌟 (evening)";
```

Comment:
For future reviewers:

Source: https://github.com/pop-os/cosmic-text/blob/1f4065c1c3399efad58841082212f7c039b58480/sample/hello.txt#L113

These translations do also appear to be correct.